### PR TITLE
[TASK] Fix issues in base.js and add more tests

### PR DESCRIPTION
**Pull request by clark800**

### CODE REVIEW

File: src/js/ripple/base.js [hunk 1]
```diff
@@ -145,16 +133,10 @@ Base.decode_check = function(version, input, alphabet) {
...
-
-    for (var i = 0, l = version.length; i < l; i++) {
-      match |= version[i] === buffer[0];
-    }
-
-    if (!match) {
-      return NaN;
-    }
+  if (Array.isArray(version) && _.every(version, function(v) {
+          return v !== buffer[0];
```
Comment (wltsmrz) [inline src/js/ripple/base.js]: This is kind of yucky

**Comment by wltsmrz** [comment]
LGTM

File: src/js/ripple/base.js (hunk 1 above)
Comment (clark800) [inline src/js/ripple/base.js]: Yes, ideally it would be written as
`if (Array.isArray(version) && _.every(version, v => v !== buffer[0]))`

Comment (wltsmrz) [inline src/js/ripple/base.js]: I wouldn't call that ideal

**Pull request merged**